File: GestionTicketsAPI/Entities/Societe.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GestionTicketsAPI.Entities
{
  public class Societe
  {
    [Key]
    public int Id { get; set; }

    [Required]
    [StringLength(100)]
    public string Nom { get; set; } = string.Empty;

    [Required]
    public string Adresse { get; set; } = string.Empty;

    [Required]
    [StringLength(50)]
    public string Telephone { get; set; } = string.Empty;

    public int PaysId { get; set; }
    public Pays? Pays { get; set; }

    public ICollection<User>? Utilisateurs { get; set; }
    public ICollection<Projet>? Projets { get; set; }


    [InverseProperty("SocietePartenaire")]
    public ICollection<Contrat> ContratsPartenaire { get; set; } = new List<Contrat>();
  }
}

File: gestion-tickets/src/app/_models/societe.ts
import { Projet } from "./Projet";
import { User } from "./user";

export interface Societe {
  id: number;
  nom: string;
  adresse: string;
  telephone: string;
  paysId: number;
  selected?: boolean;
  utilisateurs: User[];
  projets: Projet[]; 
  contrat?: any
}
